modules/vaos/app/controllers/vaos/v2/appointments_controller.rb

```diff
@@ -46,21 +64,47 @@ def appointment
           appointments_service.get_appointment(appointment_id)
       end
 
-      def update_appt_id
-        params.require(:id)
+      def new_appointment
+        @new_appointment ||=
+          appointments_service.post_appointment(create_params)
       end
 
-      def status_update
-        params.require(:status)
+      def updated_appointment
+        @updated_appointment ||=
+          appointments_service.update_appointment(update_appt_id, status_update)
+      end
+
+      def merge_clinic_names(appointments)
+        cached_clinic_names = {}
+        appointments.each do |appt|
+          unless appt[:clinic].nil?
+            unless cached_clinic_names[:clinic]
+              clinic_name = get_clinic_name(appt[:location_id], appt[:clinic])
+              cached_clinic_names[appt[:clinic]] = clinic_name
+            end
+
+            appt[:station_name] = cached_clinic_names[appt[:clinic]] if cached_clinic_names[appt[:clinic]]
+          end
+        end
       end
 
-      def cancellation_reason
-        params.require(:reason)
+      def get_clinic_name(location_id, clinic_id)
+        clinics = systems_service.get_facility_clinics(location_id: location_id, clinic_ids: clinic_id)
+        clinics.first[:station_name]
```

Comment:
so the assumption is that the only clinic that you're interested in will be the first one returned?

Reply:
That is correct, with this particular solution we will request a single clinic at a time so only a single clinic should be returned in the arry.